code.js

```diff
@@ -1,3 +1,51 @@
-function permutationSort(a) {
-    return 0;
+function permutationSort(array) {
+  var permutationsTried = 0; // declare permutations
+
+  if (array.length === 0) {
+      return permutationsTried; // base case for empty array
+  }
+
+  function genPermutations(array, currentIndex) {  
+      if (currentIndex === array.length - 1) { // base case to see if it has reached last element
+          permutationsTried++; // increments permutations as it goes through
+          if (isSorted(array)) {
+              return true; // if latest permutation has it all sorted, returns true 
+          }
+      }
+
+      for (let i = currentIndex; i < array.length; i++) { // iterates through the array 
+          swap(array, i, currentIndex); 
+          if (genPermutations(array, currentIndex + 1)) { // makes more permutations 
+              return true; 
+          }
+          swap(array, i, currentIndex); 
+      }
+      return false;  
+  }
+
+  function isSorted(array) {  
+      for (let i = 0; i < array.length - 1; i++) {
+          if (array[i] > array[i + 1]) { // checks to see if any elements are out of order
+              return false; 
+          }
+      }
+      return true; 
+  }
+
+  function swap(array, i, j) { 
+      const temp = array[i]; // temp variable for swapping later
+      array[i] = array[j]; // swaps value at j 
+      array[j] = temp; // swaps value at i 
+  }
+
+  genPermutations(array, 0); 
+
+  console.log("The total amount of permutations tried was " + permutationsTried);
```

Comment:
Please don't leave debug output in the code.

Reply:
Got it. I noticed I forgot my plagiarism note there as well.